packages/components/src/popover/limit-shift.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * External dependencies
+ */
+import type {
+	Axis,
+	Coords,
+	Placement,
+	Side,
+	MiddlewareArguments,
+} from '@floating-ui/react-dom';
+
+/**
+ * Custom limiter function for the `shift` middleware.
```

Comment:
We might need some more comments [like here](https://github.com/WordPress/gutenberg/pull/44112/files#diff-b1eae6bea5fe8fa3c1b81d05c659aa377dd3f7c6f90fd8ad3a41e0ec178c0afdR1)

Reply:
Good shout! Done in https://github.com/WordPress/gutenberg/pull/42950/commits/c2925087bebf3fbec0e56c304d6083cea7f9b247